core/src/main/scala/fs2/async/mutable/Signal.scala

```diff
@@ -26,6 +26,11 @@ trait Signal[F[_], A] extends immutable.Signal[F, A] { self =>
   def modify(f: A => A): F[Async.Change[A]]
 
   /**
+    *  like `modify` but allows to extract `B` from `A` and return it together with Change
+    */
+  def modify2[B](f: A => (A,B)):F[(Async.Change[A], B)]
```

Comment:
perhaps provide default implementation of `modify` in terms of `modify2` + `map`?